samples/snippets/src/main/java/com/example/spanner/CreateBackupWithEncryptionKey.java

```diff
@@ -0,0 +1,119 @@
+import com.google.cloud.spanner.encryption.CustomerManagedEncryption;
+import com.google.cloud.spanner.encryption.EncryptionConfigs;
+import com.google.spanner.admin.database.v1.CreateBackupMetadata;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import org.threeten.bp.LocalDateTime;
+import org.threeten.bp.OffsetDateTime;
+
+public class CreateBackupWithEncryptionKey {
+
+  static void createBackupWithEncryptionKey() throws InterruptedException {
+    // TODO(developer): Replace these variables before running the sample.
+    String projectId = "my-project";
+    String instanceId = "my-instance";
+    String databaseId = "my-database";
+    String backupId = "my-backup";
+    String kmsKeyName = "my-key";
+
+    try (Spanner spanner =
+        SpannerOptions.newBuilder().setProjectId(projectId).build().getService()) {
+      DatabaseAdminClient adminClient = spanner.getDatabaseAdminClient();
+      createBackupWithEncryptionKey(
+          adminClient,
+          projectId,
+          instanceId,
+          databaseId,
+          backupId,
+          kmsKeyName);
+    }
+  }
+
+  static Void createBackupWithEncryptionKey(DatabaseAdminClient adminClient,
+      String projectId, String instanceId, String databaseId, String backupId, String kmsKeyName)
+      throws InterruptedException {
+    // Set expire time to 14 days from now.
+    final Timestamp expireTime = Timestamp.ofTimeMicroseconds(TimeUnit.MICROSECONDS.convert(
+        System.currentTimeMillis() + TimeUnit.DAYS.toMillis(14), TimeUnit.MILLISECONDS));
+    final Backup backupToCreate = adminClient
+        .newBackupBuilder(BackupId.of(projectId, instanceId, backupId))
+        .setDatabase(DatabaseId.of(projectId, instanceId, databaseId))
+        .setExpireTime(expireTime)
+        .setEncryptionConfig(EncryptionConfigs.customerManagedEncryption(kmsKeyName))
+        .build();
+    final OperationFuture<Backup, CreateBackupMetadata> operation = adminClient
+        .createBackup(backupToCreate);
+
+    Backup backup;
+    try {
+      System.out.println("Waiting for operation to complete...");
+      backup = operation.get(1200, TimeUnit.SECONDS);
+    } catch (ExecutionException e) {
+      // If the operation failed during execution, expose the cause.
+      throw (SpannerException) e.getCause();
+    } catch (InterruptedException e) {
+      // Throw when a thread is waiting, sleeping, or otherwise occupied,
+      // and the thread is interrupted, either before or during the activity.
+      throw SpannerExceptionFactory.propagateInterrupt(e);
+    } catch (TimeoutException e) {
+      // If the operation timed out propagates the timeout
+      throw SpannerExceptionFactory.propagateTimeout(e);
+    }
+
+    // Waits for the backup to be ready
+    int maxWaitRetries = 10;
+    int currentRetry = 0;
+    while (currentRetry < maxWaitRetries && !backup.isReady()) {
```

Comment:
Is this necessary, considering the call to `operation.get(1200, TimeUnit.SECONDS)` above? I would expect the backup to be ready once the operation has finished.